src/CMakeLists.txt

```diff
@@ -35,58 +35,87 @@ else()
     message(FATAL_ERROR "Operating system undefined.")
 endif()
 
-find_package(OpenSSL REQUIRED)
-
-include(GenerateExportHeader)
-
-add_library(mgclient-static STATIC ${mgclient_src_files})
-set_target_properties(mgclient-static PROPERTIES
-        OUTPUT_NAME mgclient)
-target_compile_definitions(mgclient-static PUBLIC MGCLIENT_STATIC_DEFINE)
-target_include_directories(mgclient-static
-        PRIVATE
-        "${PROJECT_SOURCE_DIR}/src"
-        PUBLIC
-        "${PROJECT_SOURCE_DIR}/include"
-        "${CMAKE_CURRENT_BINARY_DIR}"
-        "${OPENSSL_INCLUDE_DIR}")
-target_link_libraries(mgclient-static
-        PRIVATE
-        ${OPENSSL_LIBRARIES} project_options project_c_warnings)
-if(MGCLIENT_ON_WINDOWS)
-    target_link_libraries(mgclient-static PUBLIC ws2_32)
+if(NOT EMSCRIPTEN)
+  find_package(OpenSSL REQUIRED)
 endif()
```

Comment:
Why not to move this `find_package` to the else branch of the following if statement?